bouncer/shared/fill_or_kill.ts

```diff
@@ -0,0 +1,139 @@
+import { InternalAsset as Asset, InternalAssets as Assets } from '@chainflip/cli';
+import assert from 'assert';
+import {
+  amountToFineAmount,
+  amountToFineAmountBigInt,
+  assetDecimals,
+  decodeDotAddressForContract,
+  isWithinOnePercent,
+  newAddress,
+  observeBalanceIncrease,
+  shortChainFromAsset,
+} from './utils';
+import { requestNewSwap } from './perform_swap';
+import { send } from './send';
+import { getBalance } from './get_balance';
+import { getChainflipApi, observeEvent } from './utils/substrate';
+import { RefundParameters } from './new_swap';
+
+/// Do a swap with unrealistic minimum price so it gets retried and then refunded.
+async function testMinPriceRefund(asset: Asset, amount: number) {
+  const swapAsset = asset === Assets.Usdc ? Assets.Flip : Assets.Usdc;
+  const refundAddress = await newAddress(asset, 'FOK_REFUND');
```

Comment:
If we initiate two swaps with the same asset this breaks because it will generate the same address for each - which is not obvious (nor stated anywhere). We can and should use randomness here - that way we can be confident that no two tested FoK swaps will have the same refund address, so they won't interfere with each other (which would be a source of flakiness).